File: app/src/main/java/com/parabola/newtone/mvp/view/AlbumView.java
package com.parabola.newtone.mvp.view;

import com.parabola.domain.model.Track;
import com.parabola.domain.settings.ViewSettingsInteractor.TrackItemView;

import java.util.List;

import moxy.MvpView;
import moxy.viewstate.strategy.alias.AddToEndSingle;
import moxy.viewstate.strategy.alias.OneExecution;

@AddToEndSingle
public interface AlbumView extends MvpView {
    void setAlbumTitle(String albumTitle);
    void setAlbumArtist(String artistName);
    void setAlbumArt(Object artCover);

    void refreshTracks(List<Track> tracks);
    void setItemViewSettings(TrackItemView itemViewSettings);
    void setItemDividerShowing(boolean showed);

    void setCurrentTrack(int trackId);

    @OneExecution
    void showToast(String toastMessage);

}

File: app/src/main/java/com/parabola/newtone/mvp/view/AlbumView.kt
package com.parabola.newtone.mvp.view

import com.parabola.domain.model.Track
import com.parabola.domain.settings.ViewSettingsInteractor.TrackItemView
import moxy.MvpView
import moxy.viewstate.strategy.alias.AddToEndSingle

@AddToEndSingle
interface AlbumView : MvpView {
    fun setAlbumTitle(albumTitle: String)
    fun setAlbumArtist(artistName: String)
    fun setAlbumArt(artCover: Any?)

    fun refreshTracks(tracks: List<Track>)
    fun setItemViewSettings(itemViewSettings: TrackItemView)
    fun setItemDividerShowing(showed: Boolean)

    fun setCurrentTrack(trackId: Int)
}
